overlord/ifacestate/handlers.go

```diff
@@ -386,7 +390,7 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) error {
 
 	var policyChecker interfaces.PolicyFunc
 
-	if autoConnect {
+	if autoConnect && !byGadget {
```

Comment:
Why !byGadget here? Can you add a comment in the code?

Reply:
Ah, ok, I got it, we don't want autoconnect checker for gadget connections. I wonder if we should set both flags (auto and by-gadget) on gadget connections then, auto was meant really to signify auto-connections.

Reply:
yes,  we want most of the behaviors as for autoconnect execept the different checker, in particular we want the same conflict checks

Reply:
added a comment before the 'if'